lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -405,24 +405,33 @@ export class NativeInteractionClient extends BaseInteractionClient {
             "NativeInteractionClient - handleNativeResponse called."
         );
 
-        if (response.account.id !== request.accountId) {
-            // User switch in native broker prompt is not supported. All users must first sign in through web flow to ensure server state is in sync
-            throw createNativeAuthError(NativeAuthErrorCodes.userSwitch);
-        }
-
-        // Get the preferred_cache domain for the given authority
-        const authority = await this.getDiscoveredAuthority(request.authority);
-
         // generate identifiers
         const idTokenClaims = AuthToken.extractTokenClaims(
             response.id_token,
             base64Decode
         );
+
         const homeAccountIdentifier = this.createHomeAccountIdentifier(
             response,
             idTokenClaims
         );
 
+        const cachedhomeAccountId =
+            this.browserStorage.getAccountInfoFilteredBy({
+                nativeAccountId: request.accountId,
+            })?.homeAccountId;
+
+        if (
+            homeAccountIdentifier != cachedhomeAccountId &&
+            response.account.id !== request.accountId
```

Comment:
Can we add the use cases in description of this PR where this is true that is, home account ID matches but account response differs and vice versa? I thought this was added to address the difference between embedded frame vs the parent frame in the request and am curious where this bug surfaced?

Reply:
Yep that's the main use case. Also, parent's native account id seemed to be getting passed on to the embeddd frame, so I have included a fix for that in this PR as well.